### PR TITLE
Article translations

### PR DESCRIPTION
**Pull request by khaledkz**
Add a language field to the Articles model and make the articlelist show the article in the English language

### CODE REVIEW

File: server/models/Article.js [hunk 1]
```diff
@@ -12,6 +12,12 @@ const schema = new Schema({
     visible: {
         type: Boolean,
         default: true,
+    },
+    language:{
+        type:String,
+        default:'En',
+        upsert: true,
```
Comment (kabaros) [inline server/models/Article.js]: why upsert: true?

File: server/public/script/admin-add-article.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+   
+ document.getElementById('batata').addEventListener('click',()=>{
```
Comment (kabaros) [inline server/public/script/admin-add-article.js]: what is this??

File: server/routes/api.js [hunk 3]
```diff
@@ -14,7 +14,7 @@ router.get('/articles', function (req, res, next) {
             res.json(articles)
         }
     };
-    articleClient.findArticles({ visible: true }, callBack)
+    articleClient.findArticles({ visible: true,language:"En" }, callBack)
```
Comment (kabaros) [inline server/routes/api.js]: could you instead try and get language from `req.params` and if nothing is provided then default to 'En'

File: server/views/admin-add-article.hbs [hunk 4]
```diff
@@ -14,11 +14,43 @@
...
+    </div>
+
+
     <div class="form-group has-success">
         <label class="text-success">Full Content:</label>
         <textarea class="form-control form-control-success mb-4" id="inputSuccess1" name="fullContent"></textarea>
-        <input class="btn btn-success" type="submit" value="Submit">
     </div>
 
+    <div class="form-group has-success">
```
Comment (kabaros) [inline server/views/admin-add-article.hbs]: you wouldn't  need the input for three languages in the same form with this setup. Just have a dropdown list where you choose the language and one field for the content (that will be for that specific language)

**Review by kabaros — CHANGES_REQUESTED** [review]

File: server/public/script/admin-add-article.js (hunk 2 above)
Comment (khaledkz) [reply server/public/script/admin-add-article.js]: sorry I added this file by mistake

Comment (khaledkz) [reply server/public/script/admin-add-article.js]: I delete it

File: server/views/admin-add-article.hbs (hunk 4 above)
Comment (khaledkz) [reply server/views/admin-add-article.hbs]: Alright I added this to test if the article list will show the article if i change the language

File: server/models/Article.js (hunk 1 above)
Comment (khaledkz) [reply server/models/Article.js]: I was trying to update the database to take default En but I think no need for upsert

File: server/routes/api.js (hunk 3 above)
Comment (khaledkz) [reply server/routes/api.js]: How can I get language from req.params in API?

Comment (kabaros) [reply server/routes/api.js]: in the url, you can send `/api/articles?language=En` - you can read it as `req.params.language`

File: server/views/admin-add-article.hbs [hunk 5]
```diff
@@ -13,12 +13,14 @@
             <option value={{this._id}}>{{ this.title }}</option>
             {{/each}}
         </select>
-    </div>
+
```
Comment (kabaros) [inline server/views/admin-add-article.hbs]: En => ENglish
Ar => Arabic
Am => Amharic

File: server/routes/api.js [hunk 6]
```diff
@@ -14,7 +14,13 @@ router.get('/articles', function (req, res, next) {
             res.json(articles)
         }
     };
-    articleClient.findArticles({ visible: true }, callBack)
+    let articleLanguage = req.query.language
+    if (articleLanguage) {
+        articleClient.findArticles({ visible: true, language: articleLanguage }, callBack)
+    }else{
+        articleClient.findArticles({ visible: true,language:'En' }, callBack)
+
```
Comment (kabaros) [inline server/routes/api.js]:
```js
let { language } = req.query;
if(!language) language = 'En';
articleClient.findArticles({ visible: true, language  }, callBack)


```

**Review by kabaros — APPROVED** [review]

**Pull request merged**